src/lib/stepper/stepper.md

```diff
@@ -0,0 +1,128 @@
+Angular Material stepper walks users through multi-step processes by breaking them up
```

Comment:
The docs pattern I've been trying to keep is something like
* One-sentence summary of the component (tabs docs seem to be the exception)
* Live overview example
* Individual sections specific to the component
* Accessibility section

The live example can go in a follow up PR